### PR TITLE
[PHP] Clarify Composer use, add dependency `composer-packages`

### PR DESCRIPTION
**Pull request by joshuatcasey**
## Summary
Clarify how Composer is used, add dependency `composer-packages` that is provided by `composer-install` and required by `php-start`.

### CODE REVIEW

File: text/php/0001-restructure.md [hunk 1]
```diff
@@ -54,15 +54,28 @@ buildpacks[<sup>2</sup>](#note-2):
...
+    and makes it available on the `$PATH`
+
+* **composer-global**:
+  Detect: When `BP_COMPOSER_INSTALL_GLOBAL` is set
+  * provides: none
+  * requires:
+    * `php` at build
+    * `composer` at build
+  Build: Runs `composer global` to ensure that global PHP packages are available on the `$PATH`
+      for composer install scripts
```
Comment (sophiewigmore) [inline text/php/0001-restructure.md:70]: What's the main rationale for having this split out from the `composer` buildpack?

Comment (joshuatcasey) [reply text/php/0001-restructure.md:70]: `composer-global` installs packages that are only meant for composer scripts, not for the PHP application itself. 

Its output should be on the `$PATH` for `composer-install` in the build container but should not be on the `$PATH` in the launch container, so I felt splitting it out would accomplish that.

**Comment by joshuatcasey** [comment]
> To me, this still doesn't address the question of whether the `composer-install` buildpack should know when the packages it's installing will be needed. How will this work in your idea?

It was called out here as an edit to PHP RFC 0001:

```markdown
* **composer-global**:
  Detect: When `BP_COMPOSER_INSTALL_GLOBAL` is set
  * provides: none

File: text/php/0001-restructure.md (hunk 1 above)
Comment (sophiewigmore) [reply text/php/0001-restructure.md:70]: Could we add it as a step in `composer-install`?

Comment (sophiewigmore) [reply text/php/0001-restructure.md:70]: I think other install buildpacks do something similar

**Comment by sophiewigmore** [comment]
@joshuatcasey I do not understand how the composer global buildpack helps `composer install` figure out when the packages installed by running `composer install` should be made available. From what you've laid out, I thought the composer global buildpack makes  other packages available for `composer` to use before running `composer install`. I am wondering about the packages that get installed by `composer install`.

**Comment by sophiewigmore** [comment]
Maybe you could just add a comment to the Composer Install buildpack line in which you mention that it's layer will always be available at launch to support the current language family use cases, which may be subject to change in the future depending on the use cases we support.

File: text/php/0001-restructure.md (hunk 1 above)
Comment (joshuatcasey) [reply text/php/0001-restructure.md:70]: Ok, it can be done inside `composer-install`.

**Review by thitch97 — APPROVED** [review]

**Review by sophiewigmore — APPROVED** [review]

**Pull request merged**